PowerShell/JumpCloud Commands Gallery/Mac Commands/Mac - Uninstall Remote Assist.md

```diff
@@ -0,0 +1,52 @@
+#### Name
+
+Mac - Uninstall Remote Assist | v1.0 JCCG
+
+#### commandType
+
+mac
+
+#### Command
+
+```
+#!/usr/bin/env bash
+set -euo pipefail
+
+function get_app_pid() {
+    local -r APP_NAME=$1
+    ps -eo pid,comm | awk "/${APP_NAME}$/ {print \$1}"
+}
+
+function remove_app_by_name() {
+    local -r APP_NAME=$1
+    if PID=$(get_app_pid "${APP_NAME}") && [[ -n "${PID}" ]]; then
+        kill -s TERM "${PID}"
+        sleep 1
+    fi
+
+    if PID=$(get_app_pid "${APP_NAME}") && [[ -n "${PID}" ]]; then
```

Comment:
I squinted hard, but what's the difference between this if and the above if?

Reply:
There is no difference between the if expressions. These are just escalating forms of the kill command i.e. if `kill -s TERM` doesn't kill the process, then `kill -s KILL`
```
     9       KILL (non-catchable, non-ignorable kill)
     15      TERM (software termination signal)
```     

This pattern was already authored by Wes in the Install command, so I re-used here.

Reply:
Hmm I was wondering why they weren't just combined, since we do SIGTERM, sleep, then SIGKILL anyway.

Reply:
ideally the SIGTERM kill command gracefully shuts down the process and we never execute the SIGKILL kill command. Something like this would likely fail on `kill -s KILL "${PID}"` because the pid no longer exists. 

```
 if PID=$(get_app_pid "${APP_NAME}") && [[ -n "${PID}" ]]; then
        kill -s TERM "${PID}"
        sleep 1
        kill -s KILL "${PID}"
    fi

Reply:
Ahh, the `-n` is key, thanks!